src/components/MultiHopTrade/hooks/useReceiveAddress.tsx

```diff
@@ -11,29 +12,36 @@ import {
   selectBuyAsset,
   selectManualReceiveAddress,
 } from 'state/slices/swappersSlice/selectors'
-import { useAppDispatch, useAppSelector } from 'state/store'
+import { useAppSelector } from 'state/store'
 
-const getReceiveAddress = async ({
-  asset,
-  wallet,
-  accountMetadata,
-}: GetReceiveAddressArgs): Promise<string | undefined> => {
-  const { chainId } = fromAssetId(asset.assetId)
-  const { accountType, bip44Params } = accountMetadata
-  const chainAdapter = getChainAdapterManager().get(chainId)
-  if (!(chainAdapter && wallet)) return
-  const { accountNumber } = bip44Params
-  try {
-    return await chainAdapter.getAddress({ wallet, accountNumber, accountType })
-  } catch (e) {
-    console.log(e)
-  }
```

Comment:
This would rug the p-memoization and never throw, meaning we would cache failures as `undefined`